pkgs/development/libraries/libxml2/default.nix

```diff
@@ -94,8 +94,10 @@ stdenv.mkDerivation rec {
 
   doCheck =
     (stdenv.hostPlatform == stdenv.buildPlatform) &&
-    !stdenv.isDarwin &&
     stdenv.hostPlatform.libc != "musl";
+  preCheck = lib.optional stdenv.isDarwin ''
+    export DYLD_LIBRARY_PATH="$PWD/.libs:$DYLD_LIBRARY_PATH"
```

Comment:
It worked for me with just `export DYLD_LIBRARY_PATH="$PWD/.libs`. Isn't that clearer, to show that there is no previous value of DYLD_LIBRARY_PATH that affects the build?

Reply:
I think it's fine this way.  For example, if some other part started to need this definition, getting it rewritten somewhere later might be surprising.